config_test.go

```diff
@@ -1,57 +1,165 @@
 package main
 
 import (
-	"path"
+	"bytes"
+	"log"
 	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
 )
 
 func Test_Load(t *testing.T) {
-	tt := []struct {
-		Path string
-		Err  error
-	}{
-		{
-			"valid_config.json",
-			nil,
-		}, {
-			"valid_config_extras.json",
-			nil,
-		}, {
-			"invalid_config_author.json",
-			ErrAuthorRequired,
-		}, {
-			"invalid_config_email.json",
-			ErrEmailRequired,
-		},
-	}
-
-	for _, tc := range tt {
-		path := path.Join("testdata", tc.Path)
-		_, err := LoadConfigFile(path)
-		if err != tc.Err {
-			t.Fatalf("Load(%v) returned an error (%v)", path, err)
+	Convey("When loading a valid config file", t, func() {
+		fs = mockFS{
```

Comment:
Don't get discouraged; we certainly need testing.

The reason you may get discouraged:

You're writing tests at a somewhat volatile time—the project is changing quickly and you are also developing features at the same time as writing tests.

Example, creating the the file system interface to abstract making calls / running os.exec. This is a smart idea, but it's also coming at the same time as tests for that feature. I would prefer if we step through adding features and their associated tests step by step so that I don't get lost and we don't "prefactor" (refactor before we need to). 

I've got to run, but you've touched on a few important points in this pr; of which at least 1 or 2 could be seperate PRs. 

writing tests on fast moving projects is challenging ;)